### ISSUE TITLE
gif generation does not finish / infinite creation of gif

### ISSUE THREAD

**Issue by Ventuquies** [leoding86/pixiv-omina#41]
gif generation does often not finish, even if I wait a very long time. It seems the image creation process hangs. The gif generation which has problems seems to get the jpg files from a zip with a json file which includes the timing and files. Is there any way to regenerate the gif if the creation fails? Is there another tool which can convert the zip with jpg and json to gif or another animated format? 

This would be great if fixed, thank you so much!
**love your software!!***

**Comment by Proutbedaine** [comment]
Yes, not only that but also the GIF generation process is extremely slow. Much more than it should I feel? It seems Omina creates 4 processes so I guess it can work with 4 core which is nice. But I have a i9-12900KF which has 16 physical core which half of those hyperthreads. So in the end I have 24 logical cores. I feel that if the app could take advantage of that, or at least a more of it than currently, it would shred through generating those files...

I just downloaded a profile that had 380+ GIF to generate. That was started at 10:30PM EST and it is now 6:40AM EST and there is still 178 jobs left still. So that means that theoretically I still have to wait about 7 more hours before it's done for a total of more or less 15hrs!!!

I hope there is a way to improve this also. And if it takes a separate software like you mentioned I'm ready to use that with the zips instead no questions asked lol. 

With that being said, I really love the app in general otherwise. The only other quirk is the fact I can't select anything else than the user download folder for the download even if I tried everything mentioned in the troubleshooting of other posts. But I can deal with that.
